### ISSUE TITLE
aks-engine rotate-certs unable to renew kube-proxy certs

### ISSUE THREAD

**Issue by jprecuch** [Azure/aks-engine#4450]
Keys on our aks-engine k8s 1.16.15 nodes expired because they were deployed long ago when default age was 2 years. We found out it is possible to renew them using aks-engine tool. 
We've tried to use it but seems after it finished job, but we can see that proxy.crt is still expired
```
root# ./aks-engine-v0.60.1-linux-amd64/aks-engine rotate-certs --api-model /root/_output/k8swestus2stagedc/apimodel.json --client-id ID --client-secret SEC --location westus2 --resource-group RG --subscription-id SUBID --auth-method=client_secret --linux-ssh-private-key /root/.ssh/id_rsa  --ssh-host PROPERIP
INFO[0005] Backing up artifacts to directory /root/_output/k8swestus2stagedc/_rotate_certs_backup
INFO[0005] Generating new certificates
INFO[0042] Writing artifacts to output directory /root/_output/k8swestus2stagedc/_rotate_certs_output
INFO[0042] Waiting for cluster nodes readiness: [k8s-master-26126023-0 k8s-master-26126023-1 k8s-master-26126023-2]
INFO[0082] Checking health of control plane components
INFO[0122] Distributing control plane certificates
INFO[0191] Backing up node certificates
INFO[0195] Rotating control plane certificates
INFO[0198] Rebooting control plane nodes
INFO[0409] Waiting for cluster nodes readiness: [k8s-master-26126023-0 k8s-master-26126023-1 k8s-master-26126023-2]
INFO[0519] Rotating proxy certificates
INFO[0536] Deleting temporary artifacts from control plane nodes
INFO[0540] Waiting for cluster nodes readiness: [k8s-master-26126023-0 k8s-master-26126023-1 k8s-master-26126023-2]
INFO[0581] Checking health of control plane components
INFO[0621] Distributing agent certificates
INFO[0686] Backing up node certificates
INFO[0704] Rotating agents certificates
INFO[0803] Deleting temporary artifacts from agent nodes
INFO[0818] Waiting for cluster nodes readiness: [k8s-node-26126023-vmss00002m k8s-node-26126023-vmss00002d k8s-node-26126023-vmss00002h k8s-node-26126023-vmss00002l k8s-node-26126023-vmss00002e k8s-node-26126023-vmss00002g k8s-node-26126023-vmss00002k k8s-node-26126023-vmss00002c k8s-node-26126023-vmss00002j k8s-node-26126023-vmss00002n k8s-node-26126023-vmss00002f k8s-node-26126023-vmss00002i]
INFO[0858] Recreating service account tokens
INFO[0859] Checking health of all kube-system pods
INFO[1569] Generating new artifacts
INFO[1569] Certificate rotation completed
```

I've checked the master nodes and the date of cert files changed
```
root@k8s-master-26126023-0:/etc/kubernetes/certs# ls -la
total 96
drwxr-xr-x 2 root root 4096 Jun  9 05:59 .
drwxr-xr-x 6 root root 4096 Jun  9 05:59 ..
-rw-r--r-- 1 root root 6043 Jun  9 05:53 apiserver.crt
-rw------- 1 root root 3243 Jun  9 05:53 apiserver.key
-rw-r--r-- 1 root root 1720 Jun  9 05:53 ca.crt
-rw------- 1 root root 3247 Jun  9 05:53 ca.key
-rw-r--r-- 1 root root 1785 Jun  9 05:53 client.crt
-rw------- 1 root root 3247 Jun  9 05:53 client.key
-rw-r--r-- 1 root root 1834 Jun  9 05:53 etcdclient.crt
-rw------- 1 root root 3243 Jun  9 05:53 etcdclient.key
-rw-r--r-- 1 etcd etcd 1830 Jun  9 05:53 etcdpeer0.crt
-rw------- 1 etcd etcd 3243 Jun  9 05:53 etcdpeer0.key
-rw-r--r-- 1 etcd etcd 1830 Jun  9 05:53 etcdpeer1.crt
-rw------- 1 etcd etcd 3243 Jun  9 05:53 etcdpeer1.key
-rw-r--r-- 1 etcd etcd 1830 Jun  9 05:53 etcdpeer2.crt
-rw------- 1 etcd etcd 3243 Jun  9 05:53 etcdpeer2.key
-rw-r--r-- 1 root root 1834 Jun  9 05:53 etcdserver.crt
-rw------- 1 etcd etcd 3243 Jun  9 05:53 etcdserver.key
-rw-r--r-- 1 root root 1147 Oct 29  2020 kubeletserver.crt
-rw------- 1 root root 1679 Oct 29  2020 kubeletserver.key
-rw-r--r-- 1 root root 1107 Jun  9 05:59 proxy-ca.crt
-rw-r--r-- 1 root root 1005 Jun  9 05:59 proxy.crt
-rw------- 1 root root 1679 Jun  9 05:59 proxy.key
```
However when we check for validity it is still old one for proxy
```
root@k8s-master-26126023-0:/etc/kubernetes/certs# openssl x509 -noout -text -in /etc/kubernetes/certs/proxy.crt | grep Not
            Not Before: Jun  5 15:46:12 2019 GMT
            Not After : Jun  4 15:46:12 2021 GMT
```
All other seems to be rotated fine
```
root@k8s-master-26126023-0:/etc/kubernetes/certs# openssl x509 -noout -text -in /etc/kubernetes/certs/apiserver.crt | grep Not
            Not Before: Jun  9 05:51:18 2021 GMT
            Not After : Jun  2 05:51:18 2051 GMT
root@k8s-master-26126023-0:/etc/kubernetes/certs# openssl x509 -noout -text -in /etc/kubernetes/certs/ca.crt | grep Not
            Not Before: Jun  9 05:51:06 2021 GMT
            Not After : Jun  2 05:51:06 2051 GMT
root@k8s-master-26126023-0:/etc/kubernetes/certs# openssl x509 -noout -text -in /etc/kubernetes/certs/client.crt | grep Not
            Not Before: Jun  9 05:51:18 2021 GMT
            Not After : Jun  2 05:51:18 2051 GMT
root@k8s-master-26126023-0:/etc/kubernetes/certs# openssl x509 -noout -text -in /etc/kubernetes/certs/etcdclient.crt | grep Not
            Not Before: Jun  9 05:51:18 2021 GMT
            Not After : Jun  2 05:51:18 2051 GMT
root@k8s-master-26126023-0:/etc/kubernetes/certs# openssl x509 -noout -text -in /etc/kubernetes/certs/etcdpeer0.crt | grep Not
            Not Before: Jun  9 05:51:18 2021 GMT
            Not After : Jun  2 05:51:18 2051 GMT
root@k8s-master-26126023-0:/etc/kubernetes/certs# openssl x509 -noout -text -in /etc/kubernetes/certs/etcdpeer1.crt | grep Not
            Not Before: Jun  9 05:51:18 2021 GMT
            Not After : Jun  2 05:51:18 2051 GMT
root@k8s-master-26126023-0:/etc/kubernetes/certs# openssl x509 -noout -text -in /etc/kubernetes/certs/etcdpeer2.crt | grep Not
            Not Before: Jun  9 05:51:18 2021 GMT
            Not After : Jun  2 05:51:18 2051 GMT
root@k8s-master-26126023-0:/etc/kubernetes/certs# openssl x509 -noout -text -in /etc/kubernetes/certs/etcdserver.crt | grep Not
            Not Before: Jun  9 05:51:18 2021 GMT
            Not After : Jun  2 05:51:18 2051 GMT
root@k8s-master-26126023-0:/etc/kubernetes/certs# openssl x509 -noout -text -in /etc/kubernetes/certs/proxy-ca.crt | grep Not
            Not Before: Jun  5 15:46:11 2019 GMT
            Not After : Jun  4 15:46:11 2024 GMT
root@k8s-master-26126023-0:/etc/kubernetes/certs# openssl x509 -noout -text -in /etc/kubernetes/certs/proxy.crt | grep Not
            Not Before: Jun  5 15:46:12 2019 GMT
            Not After : Jun  4 15:46:12 2021 GMT
```
Is there some specific case why this happened? Is there an easy way how to rotate this certs properly? 

Thanks

**Comment by jprecuch** [comment]
Adding additional logs if they will help debug this somehow more.
```
+ export WD=/etc/kubernetes/rotate-certs
+ WD=/etc/kubernetes/rotate-certs
+ export NEW_CERTS_DIR=/etc/kubernetes/rotate-certs/certs
+ NEW_CERTS_DIR=/etc/kubernetes/rotate-certs/certs
+ backup
+ '[' '!' -d /etc/kubernetes/certs.bak ']'
+ cp -rp /etc/kubernetes/certs/ /etc/kubernetes/certs.bak
+ export WD=/etc/kubernetes/rotate-certs
+ WD=/etc/kubernetes/rotate-certs
+ export NEW_CERTS_DIR=/etc/kubernetes/rotate-certs/certs
+ NEW_CERTS_DIR=/etc/kubernetes/rotate-certs/certs
+ cp_certs
+ cp -p /etc/kubernetes/rotate-certs/certs/etcdpeer0.crt /etc/kubernetes/rotate-certs/certs/etcdpeer0.key /etc/kubernetes/rotate-certs/certs/etcdpeer1.crt /etc/kubernetes/rotate-certs/certs/etcdpeer1.key /etc/kubernetes/rotate-certs/certs/etcdpeer2.crt /etc/kubernetes/rotate-certs/certs/etcdpeer2.key /etc/kubernetes/certs/
+ cp -p /etc/kubernetes/rotate-certs/certs/etcdclient.crt /etc/kubernetes/rotate-certs/certs/etcdclient.key /etc/kubernetes/certs/
+ cp -p /etc/kubernetes/rotate-certs/certs/etcdserver.crt /etc/kubernetes/rotate-certs/certs/etcdserver.key /etc/kubernetes/certs/
+ cp -p /etc/kubernetes/rotate-certs/certs/ca.crt /etc/kubernetes/rotate-certs/certs/ca.key /etc/kubernetes/certs/
+ cp -p /etc/kubernetes/rotate-certs/certs/client.crt /etc/kubernetes/rotate-certs/certs/client.key /etc/kubernetes/certs/
+ cp -p /etc/kubernetes/rotate-certs/certs/apiserver.crt /etc/kubernetes/rotate-certs/certs/apiserver.key /etc/kubernetes/certs/
+ cp -p /etc/kubernetes/rotate-certs/certs/kubeconfig /home/ubuntu/.kube/config
+ rm -f /var/lib/kubelet/pki/kubelet-client-current.pem
+ export WD=/etc/kubernetes/rotate-certs
+ WD=/etc/kubernetes/rotate-certs
+ export NEW_CERTS_DIR=/etc/kubernetes/rotate-certs/certs
+ NEW_CERTS_DIR=/etc/kubernetes/rotate-certs/certs
+ cp_proxy
+ source /etc/environment
++ ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
++ ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
++ ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
++ ETCDCTL_ENDPOINTS=https://127.0.0.1:2379
++ PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
+ /etc/kubernetes/generate-proxy-certs.sh
Generating RSA private key, 2048 bit long modulus (2 primes)
............+++++
.................................................................................................+++++
e is 65537 (0x010001)
Can't load /home/ubuntu/.rnd into RNG
140561710215616:error:2406F079:random number generator:RAND_load_file:Cannot open file:../crypto/rand/randfile.c:88:Filename=/home/ubuntu/.rnd
Generating RSA private key, 2048 bit long modulus (2 primes)
...+++++
.........................+++++
e is 65537 (0x010001)
Can't load /home/ubuntu/.rnd into RNG
140457827553728:error:2406F079:random number generator:RAND_load_file:Cannot open file:../crypto/rand/randfile.c:88:Filename=/home/ubuntu/.rnd
Signature ok
subject=CN = aggregator, O = system:masters
Getting CA Private Key
2021-06-09 05:21:35.408524 W | pkg/flags: unrecognized environment variable ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
2021-06-09 05:21:35.408598 W | pkg/flags: recognized environment variable ETCDCTL_ENDPOINTS, but unused: shadowed by corresponding flag
2021-06-09 05:21:35.408607 W | pkg/flags: unrecognized environment variable ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
2021-06-09 05:21:35.408651 W | pkg/flags: unrecognized environment variable ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
https://127.0.0.1:2379 is healthy: successfully committed proposal: took = 73.837846ms
2021-06-09 05:21:35.531262 W | pkg/flags: unrecognized environment variable ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
2021-06-09 05:21:35.531368 W | pkg/flags: recognized environment variable ETCDCTL_ENDPOINTS, but unused: shadowed by corresponding flag
2021-06-09 05:21:35.531376 W | pkg/flags: unrecognized environment variable ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
2021-06-09 05:21:35.531383 W | pkg/flags: unrecognized environment variable ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
2021-06-09 05:21:35.622581 W | pkg/flags: unrecognized environment variable ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
2021-06-09 05:21:35.622894 W | pkg/flags: recognized environment variable ETCDCTL_ENDPOINTS, but unused: shadowed by corresponding flag
2021-06-09 05:21:35.623052 W | pkg/flags: unrecognized environment variable ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
2021-06-09 05:21:35.623196 W | pkg/flags: unrecognized environment variable ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
2021-06-09 05:21:35.802495 W | pkg/flags: unrecognized environment variable ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
2021-06-09 05:21:35.803095 W | pkg/flags: recognized environment variable ETCDCTL_ENDPOINTS, but unused: shadowed by corresponding flag
2021-06-09 05:21:35.803372 W | pkg/flags: unrecognized environment variable ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
2021-06-09 05:21:35.803545 W | pkg/flags: unrecognized environment variable ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
2021-06-09 05:21:35.987615 W | pkg/flags: unrecognized environment variable ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
2021-06-09 05:21:35.988323 W | pkg/flags: recognized environment variable ETCDCTL_ENDPOINTS, but unused: shadowed by corresponding flag
2021-06-09 05:21:35.988485 W | pkg/flags: unrecognized environment variable ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
2021-06-09 05:21:35.988609 W | pkg/flags: unrecognized environment variable ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
2021-06-09 05:21:36.091541 W | pkg/flags: unrecognized environment variable ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
2021-06-09 05:21:36.091618 W | pkg/flags: recognized environment variable ETCDCTL_ENDPOINTS, but unused: shadowed by corresponding flag
2021-06-09 05:21:36.091627 W | pkg/flags: unrecognized environment variable ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
2021-06-09 05:21:36.091822 W | pkg/flags: unrecognized environment variable ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
2021-06-09 05:21:36.167092 W | pkg/flags: unrecognized environment variable ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
2021-06-09 05:21:36.167165 W | pkg/flags: recognized environment variable ETCDCTL_ENDPOINTS, but unused: shadowed by corresponding flag
2021-06-09 05:21:36.167174 W | pkg/flags: unrecognized environment variable ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
2021-06-09 05:21:36.167180 W | pkg/flags: unrecognized environment variable ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
2021-06-09 05:21:36.290608 W | pkg/flags: unrecognized environment variable ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
2021-06-09 05:21:36.290908 W | pkg/flags: recognized environment variable ETCDCTL_ENDPOINTS, but unused: shadowed by corresponding flag
2021-06-09 05:21:36.291148 W | pkg/flags: unrecognized environment variable ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
2021-06-09 05:21:36.291280 W | pkg/flags: unrecognized environment variable ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
+ export WD=/etc/kubernetes/rotate-certs
+ WD=/etc/kubernetes/rotate-certs
+ export NEW_CERTS_DIR=/etc/kubernetes/rotate-certs/certs
+ NEW_CERTS_DIR=/etc/kubernetes/rotate-certs/certs
+ cleanup
+ rm -rf /etc/kubernetes/rotate-certs
+ rm -rf /etc/kubernetes/certs.bak
+ export WD=/etc/kubernetes/rotate-certs
+ WD=/etc/kubernetes/rotate-certs
+ export NEW_CERTS_DIR=/etc/kubernetes/rotate-certs/certs
+ NEW_CERTS_DIR=/etc/kubernetes/rotate-certs/certs
+ backup
+ '[' '!' -d /etc/kubernetes/certs.bak ']'
+ cp -rp /etc/kubernetes/certs/ /etc/kubernetes/certs.bak
+ export WD=/etc/kubernetes/rotate-certs
+ WD=/etc/kubernetes/rotate-certs
+ export NEW_CERTS_DIR=/etc/kubernetes/rotate-certs/certs
+ NEW_CERTS_DIR=/etc/kubernetes/rotate-certs/certs
+ cp_certs
+ cp -p /etc/kubernetes/rotate-certs/certs/etcdpeer0.crt /etc/kubernetes/rotate-certs/certs/etcdpeer0.key /etc/kubernetes/rotate-certs/certs/etcdpeer1.crt /etc/kubernetes/rotate-certs/certs/etcdpeer1.key /etc/kubernetes/rotate-certs/certs/etcdpeer2.crt /etc/kubernetes/rotate-certs/certs/etcdpeer2.key /etc/kubernetes/certs/
+ cp -p /etc/kubernetes/rotate-certs/certs/etcdclient.crt /etc/kubernetes/rotate-certs/certs/etcdclient.key /etc/kubernetes/certs/
+ cp -p /etc/kubernetes/rotate-certs/certs/etcdserver.crt /etc/kubernetes/rotate-certs/certs/etcdserver.key /etc/kubernetes/certs/
+ cp -p /etc/kubernetes/rotate-certs/certs/ca.crt /etc/kubernetes/rotate-certs/certs/ca.key /etc/kubernetes/certs/
+ cp -p /etc/kubernetes/rotate-certs/certs/client.crt /etc/kubernetes/rotate-certs/certs/client.key /etc/kubernetes/certs/
+ cp -p /etc/kubernetes/rotate-certs/certs/apiserver.crt /etc/kubernetes/rotate-certs/certs/apiserver.key /etc/kubernetes/certs/
+ cp -p /etc/kubernetes/rotate-certs/certs/kubeconfig /home/ubuntu/.kube/config
+ rm -f /var/lib/kubelet/pki/kubelet-client-current.pem
+ export WD=/etc/kubernetes/rotate-certs
+ WD=/etc/kubernetes/rotate-certs
+ export NEW_CERTS_DIR=/etc/kubernetes/rotate-certs/certs
+ NEW_CERTS_DIR=/etc/kubernetes/rotate-certs/certs
+ cp_proxy
+ source /etc/environment
++ ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
++ ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
++ ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
++ ETCDCTL_ENDPOINTS=https://127.0.0.1:2379
++ PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
+ /etc/kubernetes/generate-proxy-certs.sh
Generating RSA private key, 2048 bit long modulus (2 primes)
......................................................................................................................+++++
.......+++++
e is 65537 (0x010001)
Can't load /home/ubuntu/.rnd into RNG
139833665429952:error:2406F079:random number generator:RAND_load_file:Cannot open file:../crypto/rand/randfile.c:88:Filename=/home/ubuntu/.rnd
Generating RSA private key, 2048 bit long modulus (2 primes)
...............................................+++++
.......................................................................................+++++
e is 65537 (0x010001)
Can't load /home/ubuntu/.rnd into RNG
139770068083136:error:2406F079:random number generator:RAND_load_file:Cannot open file:../crypto/rand/randfile.c:88:Filename=/home/ubuntu/.rnd
Signature ok
subject=CN = aggregator, O = system:masters
Getting CA Private Key
2021-06-09 05:59:56.603274 W | pkg/flags: unrecognized environment variable ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
2021-06-09 05:59:56.603604 W | pkg/flags: recognized environment variable ETCDCTL_ENDPOINTS, but unused: shadowed by corresponding flag
2021-06-09 05:59:56.603789 W | pkg/flags: unrecognized environment variable ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
2021-06-09 05:59:56.603929 W | pkg/flags: unrecognized environment variable ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
https://127.0.0.1:2379 is healthy: successfully committed proposal: took = 131.359364ms
2021-06-09 05:59:56.790009 W | pkg/flags: unrecognized environment variable ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
2021-06-09 05:59:56.790254 W | pkg/flags: recognized environment variable ETCDCTL_ENDPOINTS, but unused: shadowed by corresponding flag
2021-06-09 05:59:56.790405 W | pkg/flags: unrecognized environment variable ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
2021-06-09 05:59:56.790626 W | pkg/flags: unrecognized environment variable ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
2021-06-09 05:59:56.979678 W | pkg/flags: unrecognized environment variable ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
2021-06-09 05:59:56.980093 W | pkg/flags: recognized environment variable ETCDCTL_ENDPOINTS, but unused: shadowed by corresponding flag
2021-06-09 05:59:56.980331 W | pkg/flags: unrecognized environment variable ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
2021-06-09 05:59:56.980506 W | pkg/flags: unrecognized environment variable ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
2021-06-09 05:59:57.161278 W | pkg/flags: unrecognized environment variable ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
2021-06-09 05:59:57.161340 W | pkg/flags: recognized environment variable ETCDCTL_ENDPOINTS, but unused: shadowed by corresponding flag
2021-06-09 05:59:57.161349 W | pkg/flags: unrecognized environment variable ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
2021-06-09 05:59:57.161356 W | pkg/flags: unrecognized environment variable ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
2021-06-09 05:59:57.238374 W | pkg/flags: unrecognized environment variable ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
2021-06-09 05:59:57.238439 W | pkg/flags: recognized environment variable ETCDCTL_ENDPOINTS, but unused: shadowed by corresponding flag
2021-06-09 05:59:57.238448 W | pkg/flags: unrecognized environment variable ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
2021-06-09 05:59:57.238454 W | pkg/flags: unrecognized environment variable ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
2021-06-09 05:59:57.430249 W | pkg/flags: unrecognized environment variable ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
2021-06-09 05:59:57.430544 W | pkg/flags: recognized environment variable ETCDCTL_ENDPOINTS, but unused: shadowed by corresponding flag
2021-06-09 05:59:57.430715 W | pkg/flags: unrecognized environment variable ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
2021-06-09 05:59:57.430839 W | pkg/flags: unrecognized environment variable ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
2021-06-09 05:59:57.591009 W | pkg/flags: unrecognized environment variable ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
2021-06-09 05:59:57.591336 W | pkg/flags: recognized environment variable ETCDCTL_ENDPOINTS, but unused: shadowed by corresponding flag
2021-06-09 05:59:57.591487 W | pkg/flags: unrecognized environment variable ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
2021-06-09 05:59:57.591603 W | pkg/flags: unrecognized environment variable ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
2021-06-09 05:59:57.688437 W | pkg/flags: unrecognized environment variable ETCDCTL_CA_FILE=/etc/kubernetes/certs/ca.crt
2021-06-09 05:59:57.688567 W | pkg/flags: recognized environment variable ETCDCTL_ENDPOINTS, but unused: shadowed by corresponding flag
2021-06-09 05:59:57.688594 W | pkg/flags: unrecognized environment variable ETCDCTL_KEY_FILE=/etc/kubernetes/certs/etcdclient.key
2021-06-09 05:59:57.688618 W | pkg/flags: unrecognized environment variable ETCDCTL_CERT_FILE=/etc/kubernetes/certs/etcdclient.crt
+ export WD=/etc/kubernetes/rotate-certs
+ WD=/etc/kubernetes/rotate-certs
+ export NEW_CERTS_DIR=/etc/kubernetes/rotate-certs/certs
+ NEW_CERTS_DIR=/etc/kubernetes/rotate-certs/certs
+ cleanup
+ rm -rf /etc/kubernetes/rotate-certs
+ rm -rf /etc/kubernetes/certs.bak
```

**Comment by jadarsie** [comment]
Thx @jprecuch, we will work on a fix for this, meanwhile check #4463 for mitigation steps.
